.github/workflows/PR.yml

```diff
@@ -0,0 +1,201 @@
+name: PR Build and Test
+
+on:
+  pull_request:
+    branches: [ 'main' ]
+
+
+jobs:
+  workflow_setup:
+    name: Setup variables 
+    runs-on: ubuntu-latest
+    outputs:
+      repo_owner: ${{ steps.repo_owner.outputs.lowercase }}
+      os_matrix: "{\"os_version\":[\"debian11\",\"debian10\",\"ubuntu20\"]}"
+      testsNeeded: ${{ steps.testsNeeded.outputs.testsNeeded }}
+    steps:
+      - name: Lowercase repo owner
+        id: repo_owner
+        run: echo "lowercase=$(echo ${{ github.repository_owner }} | tr \"[:upper:]\" \"[:lower:]\")" >>$GITHUB_OUTPUT
+        shell: bash
+      - name: testsNeeded
+        id: testsNeeded
+        uses: dorny/paths-filter@v2
+        with:
+          filters: |
+            testsNeeded:
+              - '**/Dockerfile**'
+              - '**.yaml'
+              - '**.yml'
+              - '**.sh'
+              - '**.py'
```

Comment:
We should add .env files here or even better reverse the logic and just ignore *.md files.